worker/worker.py

```diff
@@ -109,6 +110,32 @@ def get_rate():
     return remaining >= math.sqrt(rate["limit"])
 
 
+RUN = None
+TASK_ID = None
+
+def heartbeat(worker_info, password, remote):
+    global ALIVE, RUN, TASK_ID
+    print("Start heartbeat")
+    payload = {"username": worker_info["username"], "password": password}
+    count = 0
+    while ALIVE:
+        time.sleep(1)
+        count += 1
+        if count == 60:
+            count = 0
+            print("Send heartbeat... ", end='')
+            payload["run_id"] = str(RUN["_id"]) if RUN else None
+            payload["task_id"] = TASK_ID
+            req = requests.post(
+                remote + "/api/beat",
+                data=json.dumps(payload),
+                headers={"Content-type": "application/json"},
+                timeout=HTTP_TIMEOUT,
+            )
+            req = json.loads(req.text)
```

Comment:
Perhaps better to use something like this to avoid a confused worker output during a network drop:
```python
            try:
                req = json.loads(req.text)
                print(req)
            except Exception as e:
                sys.stderr.write("Exception from calling heartbeat:\n")
                print(e)
```